modules/enrichmentLiftMeasurement/index.js

```diff
@@ -0,0 +1,135 @@
+      modules = internals.getCalculatedSubmodules();
+      storeTestConfig(testRun, modules, storeSplits, storageManager);
+    } else {
+      modules = testConfig.modules
+    }
+  }
+
+  modules = modules ?? internals.getCalculatedSubmodules();
+
+  const bannedModules = new Set(modules.filter(({enabled}) => !enabled).map(({name}) => name));
+  if (bannedModules.size) {
+    const init = suppression === suppressionMethod.SUBMODULES;
+    rules.push(registerActivityControl(ACTIVITY_ENRICH_EIDS, MODULE_NAME, userIdSystemBlockRule(bannedModules, init)));
+  }
+
+  if (testRun) {
+    setAnalyticLabels({[testRun]: modules});
+  }
+}
+
+export function reset() {
+  rules.forEach(unregister => unregister());
+  setAnalyticLabels({});
+  rules = [];
+}
+
+export function compareConfigs(old, current) {
+  const {modules: newModules, testRun: newTestRun} = current;
+  const {modules: oldModules, testRun: oldTestRun} = old;
+
+  const getModulesObject = (modules) => modules.reduce((acc, curr) => ({...acc, [curr.name]: curr.percentage}), {});
+
+  const percentageEqual = deepEqual(
+    getModulesObject(oldModules),
+    getModulesObject(newModules)
+  );
+
+  const testRunEqual = newTestRun === oldTestRun;
+  return percentageEqual && testRunEqual;
+}
+
+function userIdSystemBlockRule(bannedModules, init) {
+  return (params) => {
+    if ((params.init ?? true) === init && params[ACTIVITY_PARAM_COMPONENT_TYPE] === MODULE_TYPE_UID && bannedModules.has(params[ACTIVITY_PARAM_COMPONENT_NAME])) {
+      return {allow: false, reason: 'disabled due to AB testing'};
+    }
+  }
+};
+
+export function getCalculatedSubmodules() {
+  const {modules = []} = moduleConfig;
+  return modules
+    .map(({name, percentage}) => {
+      const enabled = Math.random() < percentage;
+      return {name, percentage, enabled}
+    });
+};
+
+export function getStoredTestConfig(storeSplits, storageManager) {
+  const [checkMethod, getMethod] = {
+    [storeSplitsMethod.SESSION_STORAGE]: [storageManager.sessionStorageIsEnabled, storageManager.getDataFromSessionStorage],
+    [storeSplitsMethod.LOCAL_STORAGE]: [storageManager.localStorageIsEnabled, storageManager.getDataFromLocalStorage],
+  }[storeSplits];
+
+  if (!checkMethod()) {
+    logError(`${MODULE_NAME} Unable to save testing module config - storage is not enabled`);
+    return null;
+  }
+
+  try {
+    return JSON.parse(getMethod(STORAGE_KEY));
+  } catch {
+    return null;
+  }
+};
+
+export function storeTestConfig(testRun, modules, storeSplits, storageManager) {
+  const [checkMethod, storeMethod] = {
+    [storeSplitsMethod.SESSION_STORAGE]: [storageManager.sessionStorageIsEnabled, storageManager.setDataInSessionStorage],
+    [storeSplitsMethod.LOCAL_STORAGE]: [storageManager.localStorageIsEnabled, storageManager.setDataInLocalStorage],
+  }[storeSplits];
+
+  if (!checkMethod()) {
+    logError(`${MODULE_NAME} Unable to save testing module config - storage is not enabled`);
```

Comment:
Is this an error? it might not even be a warning if memory is one of the supported methods.

Reply:
You are correct - sorry about that!